src/main/java/net/sf/rails/ui/swing/AutoSaveLoadDialog.java

```diff
@@ -125,12 +126,7 @@ private GridBagConstraints constraints(int gridx, int gridy, int leftinset,
 
     public void actionPerformed(ActionEvent arg0) {
         if (arg0.getSource().equals(okButton)) {
-            for (int i = 0; i < NUM_OPTIONS; i++) {
-                if (choiceButtons[i].isSelected()) {
-                    status = i;
-                    break;
-                }
-            }
+            status = IntStream.range(0, NUM_OPTIONS).filter(i -> choiceButtons[i].isSelected()).findFirst().orElse(status);
```

Comment:
Use of streams with logical operations (filtering) to pull out specific items (in this case first)

Reply:
This is fine in my opinion, but if we need to add more steps to the stream the expression becomes hard to read, therefore I would suggest to add some newlines to the expression. For example like this:

```java
status = IntStream.range(0, NUM_OPTIONS)
   .filter(i -> choiceButtons[i].isSelected())
   .findFirst().orElse(status);
```

Reply:
Yep completely agree, see my above about use of automatic formatting. I didn't format any of the example here as I use automatic formatting and that would have changed the whole darn file and no one would have been able to find the example in all of the format changes!

Reply:
Another note-- this is a good example why I think it's important that we have formatting rules "embedded" into the project-- I've got a set of IntelliJ files that I'll be PRing that include code style formatting rules so that everyone that is using IntelliJ will get the same set of rules. Would be nice if there was a way we could convert those rules to the Eclipse code formatter and add those files to the repo as then anyone using Eclipse OR IntelliJ would apply the same. I know I did the opposite a long long time ago when I migrated from Eclipse to IntelliJ but that was a by-hand project.. ie I took a file that had been formatted with my Eclipse rules, opened it in IntelliJ and started tweaking the rules one-by-one until executing code format in IntelliJ made no changes to the file.

Reply:
The question is where and how should we apply these rules? I think the best approach is to use a linter that checks for violations of the codestyle rules.

Reply:
You are referring to code formatting specifically? That is why I build these rules into IntelliJ as rejecting a PR post submission because of style is a waste of everyones time-- build the formatting into the developers process instead of as an afterthought that then get rejected and becomes a painful fix-it-later problem.

BUT even with that manual focus mentioned, yes, adding in a tool to check for not only style violations but also coding/logic violations is a key component to keeping the code base clean and maintainable. Many of the projects I work on make use of SonarQube for this exact scenario and that will automatically will reject a PR it fails a certain threshold. Keep in mind though that many tools, Sonar, CheckStyle, FindBugs, PMD, etc won't have rules for all of code conventions that you can configure in IDEs like Eclipse and IntelliJ so some stuff will always slip through hence why it's important to get a tool in place like the IDE with the rules setup so a developer just gets in the habit of hitting the format button when they save.

I ran Rails through my instance of SonarQube (not SonarCloud as it doesn't support some of the automation and plugins I use and does not have the deeper rule set I prefer) which is where much of these changes came from. BUT I also started to set up this repo in SonarCloud but realized this morning I don't have project permissions here to trigger a build-- another weird SonarCloud thing. In any case, it would be worth while pursuing this automated scanning setup and feeding it into a PR even if we don't use it to reject/accept...

Reply:
> You are referring to code formatting specifically? That is why I build these rules into IntelliJ as rejecting a PR post submission because of style is a waste of everyones time

I agree it would be best if everyone would use their IDE to format his/her code before commiting. The problem is that there are a lot of people that simply don't do that. Even worse there are also a not too small number of people that don't use a classic IDE but instead some other kind of editor like VS Code. I know you can also configure code formatting for such editors but every supported editor/IDE requires its own set of configuration, which makes it unfeasible. 

Another reason for rejecting a PR based on style violations is that you can often only get hold of a developer until their PR has been merged. Afterwards they often move on, either to another issue or to another project. So if you don't want to refactor your code after every three or four merged PRs then you should push your developers to provide clean, well formatted and documented code.

Reply:
Take a look at the Sonar changes PR